src/main/java/org/zalando/baigan/service/ConfigurationParser.java

```diff
@@ -0,0 +1,83 @@
+package org.zalando.baigan.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+import org.zalando.baigan.model.Condition;
+import org.zalando.baigan.model.Configuration;
+import org.zalando.baigan.proxy.ConfigTypeProvider;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.lang.reflect.Type;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toSet;
+
+@Component
+public class ConfigurationParser {
+
+    private final Logger LOG = LoggerFactory
+            .getLogger(ConfigurationParser.class);
+    final ObjectMapper objectMapper;
+    final ConfigTypeProvider configTypeProvider;
+
+    @Autowired
+    public ConfigurationParser(final ConfigTypeProvider configTypeProvider, @Qualifier("baiganObjectMapper") final Optional<ObjectMapper> objectMapper) {
```

Comment:
Do we have a test (or shall we add one) that verifies that an application context offering a specifically qualified `ObjectMapper` bean is handing it to the `ConfigurationParser` (plus the negative variants of that validation)?

Reply:
There is a test that does it, but since it does not special configuration of the object mapper, it does not actually validate it properly. I'll extend this to cover the case.